lib/turbograft.rb

```diff
@@ -36,7 +36,7 @@ def referer
 
       ActiveSupport.on_load(:action_view) do
         (ActionView::RoutingUrlFor rescue ActionView::Helpers::UrlHelper).module_eval do
-          include XHRUrlFor
+          prepend XHRUrlFor
```

Comment:
**TL, DR: ** `prepend` was added in Ruby 2, removing the need for `alias_method_chain`, which will be removed in Rails 5.

--- 
Say you want collections in Ruby  `module Identifiable` that gets included in other classes to extend the functionality the `id` method to also not return the name of the class.

```ruby
module Identifiable
  def id
    self.class_name + "." + super
  end
end

class User
  include Identifiable

  def id 
    primary_key
  end
end
```

## `include`

If you did this this way, Ruby will make `Identifiable` an ancestor of `User`. The ancestor chain of `User` would be 
```
[User, Identifiable, ...]
```

Notice `User` is first in the chain, thus calling `user.id` would simply return the primary key. Because of this, Rails implemented `alias_method_chain`, which you can use similarly to `alias_method_chain :id, :class_name`. 

This will do two things:
1 ) Rename the method `id` in `User` to `id_without_class_name`
2) Define a new `id` method like such:

```ruby
def id
  id_with_class_name
end
```

You would then define the `id_with_class_name` method in the inherited module (`Identifiable`) to do what you want, and instead of calling `super`, you would call `id_without_class_name`. This was basically a way to inject a more precise inheritance in the chain of the object when the language didn't support it. The dispatch diagram looks like something like this

```
+--------+          +----------------+
|        |          |                |
|  User  |          |  Identifiable  |
|        |          |                |
+--------+          +----------------+

  id  +------------->  id_with_class_name
                             +
                             |
  id_without_class_name <----+
  (originally called id)
```

## `prepend`

Using `prepend` flips things over. It prepends the module in the chain, so you get 

```ruby
[Identifiable, User, ...]
```

When you call `id`, it will hit the definition in `Identifiable` first, which calls `super`, which then finds the definition in `User`, resulting in the intended behaviour. 

The dispatch is much simpler to grok and debug

```
+---------------+   +--------+
|               |   |        |
| Identifiable  |   |  User  |
|               |   |        |
+---------------+   +--------+

  id   +----------->  id
```